internal/services/containers/container_group_resource.go

```diff
@@ -686,23 +714,8 @@ func resourceContainerGroupCreate(d *pluginsdk.ResourceData, meta interface{}) e
 		if dnsNameLabel := d.Get("dns_name_label").(string); dnsNameLabel != "" {
 			containerGroup.Properties.IPAddress.DnsNameLabel = &dnsNameLabel
 		}
-	}
-
-	// https://docs.microsoft.com/en-us/azure/container-instances/container-instances-vnet#virtual-network-deployment-limitations
-	// https://docs.microsoft.com/en-us/azure/container-instances/container-instances-vnet#preview-limitations
-	if networkProfileID := d.Get("network_profile_id").(string); networkProfileID != "" {
-		id, _ := networkParse.NetworkProfileID(networkProfileID)
-		networkProfileIDNorm := id.ID()
-		// Avoid parallel provisioning if "network_profile_id" is given.
-		// See: https://github.com/hashicorp/terraform-provider-azurerm/issues/15025
-		locks.ByID(networkProfileIDNorm)
```

Comment:
Shall we still keep this locking (but against the subnet, if any) to avoid issues like https://github.com/hashicorp/terraform-provider-azurerm/issues/15025?

Reply:
I think it will not hurt to have this in place.